src/ServiceControl.Audit.Persistence.PostgreSQL/PostgreSQLAuditDataStore.cs

```diff
@@ -0,0 +1,265 @@
+
+namespace ServiceControl.Audit.Persistence.PostgreSQL;
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.IO;
+using Npgsql;
+using NServiceBus;
+using ServiceControl.Audit.Auditing;
+using ServiceControl.Audit.Auditing.MessagesView;
+using ServiceControl.Audit.Infrastructure;
+using ServiceControl.Audit.Monitoring;
+using ServiceControl.Audit.Persistence;
+using ServiceControl.Audit.Persistence.Infrastructure;
+using ServiceControl.SagaAudit;
+
+
+class PostgreSQLAuditDataStore(PostgreSQLConnectionFactory connectionFactory) : IAuditDataStore
+{
+    static readonly RecyclableMemoryStreamManager manager = new RecyclableMemoryStreamManager();
+
+    public async Task<MessageBodyView> GetMessageBody(string messageId, CancellationToken cancellationToken)
+    {
+        await using var conn = await connectionFactory.OpenConnection(cancellationToken);
+        await using var cmd = new NpgsqlCommand(@"
+                select headers, body from processed_messages
+                where message_id = @message_id
+                LIMIT 1;", conn);
+        cmd.Parameters.AddWithValue("message_id", messageId);
+        await using var reader = await cmd.ExecuteReaderAsync(System.Data.CommandBehavior.SequentialAccess, cancellationToken);
+        if (await reader.ReadAsync(cancellationToken))
+        {
+            var contentType = reader.GetFieldValue<Dictionary<string, string>>(reader.GetOrdinal("headers")).GetValueOrDefault(Headers.ContentType, "text/xml");
+            using var stream = await reader.GetStreamAsync(reader.GetOrdinal("body"), cancellationToken);
+            var responseStream = manager.GetStream();
+            await stream.CopyToAsync(responseStream, cancellationToken);
+            responseStream.Position = 0;
+            return MessageBodyView.FromStream(responseStream, contentType, (int)stream.Length, string.Empty);
```

Comment:
I couldn't figure out a way to pass the stream down the pipeline without allocating.
It's possible that our abstraction is preventing it, or perhaps it's me.

@SzymonPobiega @danielmarbach any thoughts?